core/src/test/java/edu/stanford/futuredata/macrobase/pipeline/BasicBatchPipelineTest.java

```diff
@@ -15,4 +20,15 @@ public void testDemoQuery() throws Exception {
         Explanation e = p.results();
         assertEquals(3.0, e.numTotal(), 1e-10);
     }
+
+    @Test
+    public void testInlineCsv() throws Exception{
+
+        FileInputStream inputStream = new FileInputStream("demo/sample_inlinecsv.json");
+        String jsonString = IOUtils.toString(inputStream);
+        PipelineConfig conf = PipelineConfig.fromJsonString(jsonString);
+        CubePipeline p = new CubePipeline(conf);
+        Explanation e = p.results();
+        assertEquals(2328375, e.numTotal(), 1e-10);
```

Comment:
This integration test is great, thanks for including it! This file was meant for just testing the BasicBatchPipeline though, please create a CubePipelineTest and move the file there. Also please create a new PipelineUtilsTest and add a small simple unit test for the json parsing specifically so we can iterate on the json encoding / parsing in the future. Otherwise this is looking good.

Reply:
Do you mean to add a unit test that tests the parsing csv data from json?

Reply:
yeah testing the json -> csv extraction logic in a new small unit test in a PipelineUtilsTest class